### PR TITLE
Update prometheus and cortex

### PR DESCRIPTION
**Pull request by replay**
* [ ] I added CHANGELOG entry for this change.
* [X] Change is not relevant to the end user.

## Changes

This updates the vendored Prometheus and Cortex versions. In the latest version of Prometheus the Querier's `.LabelValues()` call now takes a matchers parameter, so this also adds that parameter to the `storepb.LabelValuesRequest` type.

### CODE REVIEW

File: pkg/query/querier.go [hunk 1]
```diff
@@ -333,18 +333,24 @@ func sortDedupLabels(set []storepb.Series, replicaLabels map[string]struct{}) {
...
+func (q *querier) LabelValues(name string, matchers ...*labels.Matcher) ([]string, storage.Warnings, error) {
 	span, ctx := tracing.StartSpan(q.ctx, "querier_label_values")
 	defer span.Finish()
 
 	// TODO(bwplotka): Pass it using the SeriesRequest instead of relying on context.
 	ctx = context.WithValue(ctx, store.StoreMatcherKey, q.storeDebugMatchers)
 
+	pbMatchers, err := storepb.PromMatchersToMatchers(matchers...)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "proxy LabelValues()")
```
Comment (yeya24) [inline pkg/query/querier.go:345]: Let's update the msg to `convert matchers` to be consistent.

File: pkg/store/storepb/rpc.proto [hunk 2]
```diff
@@ -178,6 +178,8 @@ message LabelValuesRequest {
   // The content of this field and whether it's supported depends on the
   // implementation of a specific store.
   google.protobuf.Any hints = 6;
+
+  repeated LabelMatcher label_matchers = 7 [(gogoproto.nullable) = false];
```
Comment (yeya24) [inline pkg/store/storepb/rpc.proto:182]: We can just call it `matchers` to be consistent.

File: pkg/query/querier.go (hunk 1 above)
Comment (bwplotka) [reply pkg/query/querier.go:345]: 👍🏽

File: pkg/store/storepb/rpc.proto (hunk 2 above)
Comment (bwplotka) [reply pkg/store/storepb/rpc.proto:182]: 👍🏽

**Review by bwplotka — APPROVED** [review]
Thank you! Some consistency fixes to address and we can merge! 💪🏽 

LGTM when fixed. Would you have time to address those comments? I can then release Thanos 0.19 with this 💪🏽

**Comment by replay** [comment]
> LGTM when fixed. Would you have time to address those comments? I can then release Thanos 0.19 with this

Great! yes I'll make the requested changes, thanks to both of you for the quick feedback.

**Comment by replay** [comment]
I have addressed the comments, please take another look

**Review by yeya24 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**